analyzer/src/test/kotlin/managers/NpmDependencyHandlerTest.kt

```diff
@@ -0,0 +1,103 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.containExactly
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+class NpmDependencyHandlerTest : StringSpec({
+    "identifierFor extracts the correct identifier" {
+        val id = createIdentifier("foo")
+        val module = createModuleInfo(id)
+
+        val handler = createHandler()
+
+        handler.identifierFor(module) shouldBe id
+    }
+
+    "dependenciesFor returns the correct dependencies" {
+        val dep1 = createModuleInfo(createIdentifier("dependency1"))
+        val dep2 = createModuleInfo(createIdentifier("dependency2"))
+        val dep3 = createModuleInfo(createIdentifier("dependency3"))
+        val dependencies = setOf(dep1, dep2, dep3)
+        val module = createModuleInfo(createIdentifier("test"), dependencies = dependencies)
+
+        val handler = createHandler()
+
+        handler.dependenciesFor(module) shouldBe dependencies
+    }
+
+    "linkageFor returns the correct package linkage" {
+        val module = createModuleInfo(createIdentifier("linkageTest"))
+
+        val handler = createHandler()
+
+        handler.linkageFor(module) shouldBe PackageLinkage.DYNAMIC
+    }
+
+    "a package can be created for a module" {
+        val pkgJsonFile = File("src/test/assets/test-package.json")
+        val module = createModuleInfo(createIdentifier("packageTest"), packageFile = pkgJsonFile)
+
+        val handler = createHandler()
+
+        val pkg = handler.createPackage(module, mutableListOf())
+
+        pkg shouldNotBeNull {
+            id shouldBe Identifier("NPM", "", "bonjour", "3.5.0")
+            declaredLicenses should containExactly("MIT")
+            authors should containExactly("Thomas Watson Steen")
+            homepageUrl shouldBe "https://github.com/watson/bonjour/local"
+            description shouldBe "A Bonjour/Zeroconf implementation in pure JavaScript (local)"
+        }
+    }
+})
+
+/** The URI to be used for queries to the NPM registry. */
+const val NPM_REGISTRY_URI = "http://npm-test.example.org"
```

Comment:
Same here, URL / ...Url instead of URI / ...Uri?

Also, even if just a dummy, I'd prefer to have "https" here instead of "http". This avoid false matches when searching the code base for places where still "http" is used.